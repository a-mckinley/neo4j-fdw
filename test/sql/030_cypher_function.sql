\a
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS ''The Matrix'' RETURN m ORDER BY id(m) DESC');
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS "The Matrix" RETURN m ORDER BY id(m) DESC');
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS $search RETURN m ORDER BY id(m) DESC','{''search'':''Matrix''}');
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS $search RETURN m ORDER BY id(m) DESC','{"search":"Matrix"}');
SELECT name, born  FROM cypher('MATCH (p:Person)-[r:ACTED_IN]->(m:Movie) WHERE m.title=$movie RETURN p.name AS name, p.born AS born ORDER BY born DESC','{"movie":"The Matrix"}') , json_to_record(cypher) as x(name varchar, born smallint);
select * from cypher('create (a {name : "test"}),(b {name : "thing"}),(c {name : "other"}) return *;');