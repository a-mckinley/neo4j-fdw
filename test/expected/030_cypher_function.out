\a
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS ''The Matrix'' RETURN m ORDER BY id(m) DESC');
cypher
{"m":{"id": 10,"labels": ["Movie"],"properties": {"tagline": "Everything that has a beginning has an end", "title": "The Matrix Revolutions", "released": 2003}}}
{"m":{"id": 9,"labels": ["Movie"],"properties": {"tagline": "Free your mind", "title": "The Matrix Reloaded", "released": 2003}}}
{"m":{"id": 0,"labels": ["Movie"],"properties": {"tagline": "Welcome to the Real World", "title": "The Matrix", "released": 1999}}}
(3 rows)
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS "The Matrix" RETURN m ORDER BY id(m) DESC');
cypher
{"m":{"id": 10,"labels": ["Movie"],"properties": {"tagline": "Everything that has a beginning has an end", "title": "The Matrix Revolutions", "released": 2003}}}
{"m":{"id": 9,"labels": ["Movie"],"properties": {"tagline": "Free your mind", "title": "The Matrix Reloaded", "released": 2003}}}
{"m":{"id": 0,"labels": ["Movie"],"properties": {"tagline": "Welcome to the Real World", "title": "The Matrix", "released": 1999}}}
(3 rows)
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS $search RETURN m ORDER BY id(m) DESC','{''search'':''Matrix''}');
cypher
{"m":{"id": 10,"labels": ["Movie"],"properties": {"tagline": "Everything that has a beginning has an end", "title": "The Matrix Revolutions", "released": 2003}}}
{"m":{"id": 9,"labels": ["Movie"],"properties": {"tagline": "Free your mind", "title": "The Matrix Reloaded", "released": 2003}}}
{"m":{"id": 0,"labels": ["Movie"],"properties": {"tagline": "Welcome to the Real World", "title": "The Matrix", "released": 1999}}}
(3 rows)
SELECT *  FROM cypher('MATCH (m:Movie)  WHERE m.title CONTAINS $search RETURN m ORDER BY id(m) DESC','{"search":"Matrix"}');
cypher
{"m":{"id": 10,"labels": ["Movie"],"properties": {"tagline": "Everything that has a beginning has an end", "title": "The Matrix Revolutions", "released": 2003}}}
{"m":{"id": 9,"labels": ["Movie"],"properties": {"tagline": "Free your mind", "title": "The Matrix Reloaded", "released": 2003}}}
{"m":{"id": 0,"labels": ["Movie"],"properties": {"tagline": "Welcome to the Real World", "title": "The Matrix", "released": 1999}}}
(3 rows)
SELECT name, born  FROM cypher('MATCH (p:Person)-[r:ACTED_IN]->(m:Movie) WHERE m.title=$movie RETURN p.name AS name, p.born AS born ORDER BY born DESC','{"movie":"The Matrix"}') , json_to_record(cypher) as x(name varchar, born smallint);
name|born
Emil Eifrem|1978
Carrie-Anne Moss|1967
Keanu Reeves|1964
Laurence Fishburne|1961
Hugo Weaving|1960
(5 rows)
select * from cypher('create (a {name : "test"}),(b {name : "thing"}),(c {name : "other"}) return *;');
cypher
{"a":{"id": 6,"labels": [],"properties": {"name": "test"}},"b":{"id": 7,"labels": [],"properties": {"name": "thing"}},"c":{"id": 8,"labels": [],"properties": {"name": "other"}}}
(1 row)
